packages/ra-core/src/controller/field/useReference.ts

```diff
@@ -84,7 +83,7 @@ export const useReference = ({
     const resourceLinkPath = !linkType
         ? false
         : typeof linkType === 'function'
-        ? linkType()
+        ? linkType(sourceId, reference)
```

Comment:
I think @fzaninotto meant the record itself

Reply:
So it should be `linkType(record)` ?

Reply:
`linkType(record, reference)` where record is the item referencing, and reference the referenced item